src/hotspot/share/gc/shenandoah/heuristics/shenandoahAdaptiveHeuristics.cpp

```diff
@@ -368,10 +367,9 @@ double ShenandoahAllocationRate::sample(size_t allocated) {
       rate = instantaneous_rate(now, allocated);
       _rate.add(rate);
       _rate_avg.add(_rate.avg());
+      _last_sample_time = now;
+      _last_sample_value = allocated;
     }
```

Comment:
This was a bug.  If we don't update the average, we should not restart the time until next sample and we should not overwrite the value of _last_sample_value.